server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -672,4 +661,146 @@ boolean waitForBackgroundTermination(long timeout) throws InterruptedException
   {
     return backgroundManagerExitedLatch.await(timeout, TimeUnit.MILLISECONDS);
   }
+
+  public static class LookupsCommunicator
```

Comment:
this is not really a new class but just moved few existing methods into it so that unit testing could be done easily. It sits in the end and doesn't really cause any understandability issues, OTOH by being in this place it gives an indication of usage within the same file only.